R/convenience-functions.R

```diff
@@ -209,3 +202,61 @@ log_shift <- function(x,
   }
   log(x + offset, base = base)
 }
+
+
+#' @title Set unit of a single forecast manually
+#'
+#' @description Helper function to set the unit of a single forecast (i.e. the
+#' combination of columns that uniquely define a single forecast) manually.
+#' This simple function keeps the columns specified in `forecast_unit` (plus
+#' additional protected columns, e.g. for true values, predictions or quantile
+#' levels) and removes duplicate rows.
+#' If not done manually, `scoringutils` attempts to determine the unit
+#' of a single automatically by simply assuming that all column names are
+#' relevant to determine the forecast unit. This can lead to unexpected
+#' behaviour, so setting the forecast unit explicitly can help make the code
+#' easier to debug and easier to read. When used as part of a workflow,
+#' `set_forecast_unit()` can then directly be piped into `check_forecasts()` to
+#' check everything is in order.
+#'
+#' @inheritParams score
+#' @param forecast_unit Character vector with the names of the columns that
+#' uniquely identify a single forecast.
+#' @return A data.table with only those columns kept that are relevant to
+#' scoring or denote the unit of a single forecast as specified by the user.
+#'
+#' @importFrom data.table ':=' is.data.table copy
+#' @author Nikos Bosse \email{nikosbosse@@gmail.com}
+#' @export
+#' @references Transformation of forecasts for evaluating predictive
+#' performance in an epidemiological context
+#' Nikos I. Bosse, Sam Abbott, Anne Cori, Edwin van Leeuwen, Johannes Bracher,
+#' Sebastian Funk
+#' medRxiv 2023.01.23.23284722
+#' \doi{https://doi.org/10.1101/2023.01.23.23284722}
+#' <https://www.medrxiv.org/content/10.1101/2023.01.23.23284722v1> # nolint
+#' @keywords data-handling
+#' @examples
+#' set_forecast_unit(
+#'   example_quantile,
+#'   c("location", "target_end_date", "target_type", "horizon", "model")
+#' )
+
+set_forecast_unit <- function(data, forecast_unit = NULL) {
+
+  datacols <- colnames(data)
+  missing <- forecast_unit[!(forecast_unit %in% datacols)]
```

Comment:
what does using the default (NULL) do exactly? Does it just end up doing the internal score behaviour of using all columns? If yes should there not be some verbose messaging to this effect?

Reply:
If this is how it behaves could it be used inside check_forecasts when the unit is determined?

Reply:
You're right, having a default NULL doesn't really make sense. As is it would return a data.frame with only the 'protected columns' (i.e. thinks like sample or quantile or true_value). I removed the default. 
`check_forecasts()` uses a function `get_forecast_unit()` instead.